sentinelhub/areas.py

```diff
@@ -427,15 +427,13 @@ def __init__(
         super().__init__(shape_list, crs, **kwargs)
```

Comment:
Github is being annoying and won't let me mark L425, `self.bbox_grid = self._parse_bbox_grid(bbox_grid)`

It is now clear that in that line we just assign input to an attribute, which is very dangerous if we are given mutable input structures. This was indirectly done before, but wasn't clear due to the `BBoxCollection` wrapper (though i described it's issues in the internal report).

Change to `list(bbox_grid)` or similar.  Check that this way its also safe to use `BBoxCollection` as input, so that we're sure we're not breaking any code (yet)

Reply:
Tested for BBoxCollection.